drivers/soundwire/intel_init.c

```diff
@@ -134,6 +136,52 @@ sdw_intel_scan_controller(struct sdw_intel_acpi_info *info)
 	return 0;
 }
 
+#define HDA_DSP_GEN_BASE                0x0
+#define HDA_DSP_REG_ADSPIC2             (HDA_DSP_GEN_BASE + 0x10)
+#define HDA_DSP_REG_ADSPIS2             (HDA_DSP_GEN_BASE + 0x14)
+
+static void sdw_intel_enable_irq(struct sdw_intel_ctx *ctx, bool enable)
+{
+	u32 val;
+
+	val = readl(ctx->mmio_base + HDA_DSP_REG_ADSPIC2);
+
+	if (enable)
+		val |= BIT(5);
+	else
+		val &= ~BIT(5);
+
+	writel(val, ctx->mmio_base + HDA_DSP_REG_ADSPIC2);
+
+}
```

Comment:
this duplicates the functionality in sound/soc/sof/intel/hda.c

so maybe we need to refactor the code and export it

Reply:
Yes, good idea.